### ISSUE TITLE
Clock Scan Problems and environments

### ISSUE THREAD

**Issue by imagina** [ianka/xfs_undelete#42]
I had to specify the timezone because clock scan was failing with: time value too large/small to represent

Adding this to the top of the scripts fixed the issue

`
set env(TZ) Europe/Kiev`

**Comment by ianka** [comment]
I wonder why Tcl cannot detect the timezone on your machine. Which OS  do you use?

**Comment by cdeviabylte** [comment]
I was running it in a rescue boot Debian 9 environment. Setting the timezone locally didn't work. Only setting the timezone directly in the script worked.

**Comment by ianka** [comment]
I have to test how to detect this problem. Could you please check if

```
#!/usr/bin/env tclsh

## Set UTC timezone if the timezone cannot be detected.
if {[catch {clock scan "now"}]} {
		set env(TZ) ":UTC"
		puts stderr "Timezone cannot be detected. Defaulting to UTC."
}
puts [clock scan "now"]
```
does the trick?

**Comment by imagina** [comment]
Yes, it's working in the rescue environment. Thank you!

**Comment by ianka** [comment]
Thanks for the extra effort to boot into the rescue environment another time.

[Fixed in v12.1](https://github.com/ianka/xfs_undelete/releases/tag/v12.1)
